File: app/src/main/java/home/oleg/popularmovies/presentation/BasicActivity.java
package home.oleg.popularmovies.presentation;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import butterknife.ButterKnife;
import home.oleg.popularmovies.R;

/**
 * Created by Oleg on 16.04.2017.
 */

public abstract class BasicActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        ButterKnife.bind(this);
        setupToolbar();
    }

    protected abstract @LayoutRes Integer getLayoutId();

    private void setupToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        if (toolbar == null) return;
        setSupportActionBar(toolbar);
    }
}

File: app/src/main/java/home/oleg/popularmovies/presentation/BasicActivity.kt
package home.oleg.popularmovies.presentation

import android.os.Bundle
import android.support.annotation.LayoutRes
import dagger.android.support.DaggerAppCompatActivity
import kotlinx.android.synthetic.main.toolbar.*

/**
 * Created by Oleg on 16.04.2017.
 */

abstract class BasicActivity : DaggerAppCompatActivity() {

    @get:LayoutRes
    protected abstract val layoutId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        setupToolbar()
    }

    private fun setupToolbar() {
        toolbar?.let(this::setSupportActionBar)
    }
}
